src/ptvsd/_vendored/pydevd/tests_python/test_debugger_json.py

```diff
@@ -2159,6 +2161,135 @@ def test_source_mapping(case_setup):
         writer.finished_ok = True
 
 
+@pytest.mark.skipif(not TEST_CHERRYPY, reason='No CherryPy available')
+def test_process_autoreload_cherrypy(case_setup_multiprocessing, tmpdir):
+    '''
+    CherryPy does an os.execv(...) which will kill the running process and replace
+    it with a new process when a reload takes place, so, it mostly works as
+    a new process connection (everything is the same except that the
+    existing process is stopped).
+    '''
+    port = get_free_port()
```

Comment:
Could just use ephemeral port 0 and parse the real one from console output...

Reply:
That's mostly what `get_free_port()` does... I know there's a (really) small chance that some other process uses the same port from that call before the server actually uses it, but I think is negligible enough to keep as is (and it'd only affect tests).